Automattic/woocommerce-payments
# Add dispute object to DisputeDetails component

Fixes #6987 

#### Changes proposed in this Pull Request



A `<DisputeDetails />` component was added in #6947. To simplify working on other tasks in the epic #6883 the `Dispute` object should be passed to this component.


**Question for the PR Reviewer:**
According to the design, there should also be a border around this component. Should this be included as well?
![image](https://github.com/Automattic/woocommerce-payments/assets/57298/142655cc-08da-4eb1-9101-b94facf43ce7)

#### Testing instructions

Adjust the `<DisputeDetails />` component to show dispute specific details. These should only show if the transaction has a dispute.

```diff
const DisputeDetails: React.FC< DisputeDetailsProps > = ( { dispute } ) => {
-	return <></>;
+       return <>{ dispute.reason }</>;
};
```

1. Enable feature flag by running `update_option( '_wcpay_feature_dispute_on_transaction_page', '1' );` in [WP Console](https://wordpress.org/plugins/wp-console/) or by modifying your database table `wp_options` directly.
2. Load a Transaction Details page of a dispute (`/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions%2Fdetails&id=pi_abc&transaction_id=txn_abc&transaction_type=charge`).
3. Make sure the `<DisputeDetails />` component is showing when viewing a disputed transaction and that it doesn't show when the transaction is not disputed.

-------------------


**Post merge**
